string directory

---

uint64 code

string message
